notifier/notification.go

```diff
@@ -5,49 +5,166 @@ import (
 	"sync"
 )
 
-// Notification types
+// Type describes the type of a notification.
+type Type uint8
+
+// Notification types.
+const (
+	Info    Type = 0
+	Warning Type = 1
+	Prompt  Type = 2
+	Error   Type = 3
+)
+
+// State describes the state of a notification.
+type State string
+
+// Possible notification states.
+// State transitions can only happen from top to bottom.
 const (
```

Comment:
Why do we have a copy of this file here instead of just importing portbase/notifications?